controller/volume_controller.go

```diff
@@ -3008,6 +3098,20 @@ func (vc *VolumeController) updateRecurringJobs(v *longhorn.Volume) (err error)
 	if err = datastore.FixupRecurringJob(v); err != nil {
 		return err
 	}
+
+	// DR volume will not restore volume recurring jobs/groups configuration.
+	if v.Spec.FromBackup != "" && !v.Spec.Standby && !v.Status.RestoreInitiated {
+		RestoreRecurringJobs, err := vc.ds.GetSettingAsBool(types.SettingNameRestoreVolumeRecurringJobs)
+		if err != nil {
+			vc.logger.WithError(err).Warnf("Failed to get %v setting", types.SettingNameRestoreVolumeRecurringJobs)
+		}
+		if v.Spec.RestoreVolumeRecurringJob == longhorn.RestoreVolumeRecurringJobEnabled || (v.Spec.RestoreVolumeRecurringJob != longhorn.RestoreVolumeRecurringJobDisabled && RestoreRecurringJobs) {
```

Comment:
You can abstract this as a separate function, e.g.,`shouldRestoreRecurringJobs()`

Reply:
Separated. Thanks.